contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--4.4.0--4.5.0.sql

```diff
@@ -10,6 +10,49 @@ SELECT set_config('search_path', 'sys, '||current_setting('search_path'), false)
  * So make sure that any SQL statement (DDL/DML) being added here can be executed multiple times without affecting
  * final behaviour.
  */
+-- Update all grants to babelfish users to make bbf_role_admin as grantor.
+DO
+LANGUAGE plpgsql
+$$
+DECLARE
+    temprow RECORD;
+    query TEXT;
+    init_user NAME;
+BEGIN
+    SELECT r.rolname
+    INTO init_user
+    FROM pg_roles r
+    INNER JOIN pg_database d
+    ON r.oid = d.datdba
+    WHERE d.datname = current_database();
+
+    FOR temprow IN
+        WITH bbf_catalog AS (
+            SELECT r.oid AS roleid, ext1.rolname
+            FROM sys.babelfish_authid_login_ext ext1
+            INNER JOIN pg_roles r ON ext1.rolname = r.rolname
+            WHERE r.rolname != init_user
+            UNION
+            SELECT r.oid AS roleid, ext2.rolname
+            FROM sys.babelfish_authid_user_ext ext2
+            INNER JOIN pg_roles r ON ext2.rolname = r.rolname
+        )
+        SELECT cat.rolname AS rolname, cat2.rolname AS member, am.grantor::regrole
+        FROM pg_auth_members am
+        INNER JOIN bbf_catalog cat ON am.roleid = cat.roleid
+        INNER JOIN bbf_catalog cat2 ON am.member = cat2.roleid
+        WHERE am.admin_option = 'f'
+        AND am.grantor != 'bbf_role_admin'::regrole
+    LOOP
+        -- First revoke the existing grant
+        query := pg_catalog.format('REVOKE %I FROM %I GRANTED BY %s;', temprow.rolname, temprow.member, temprow.grantor);
```

Comment:
I wonder if this is prone to injection. Does EXECUTE check if only 1 statement is being executed or not?